### PR TITLE
Add biases

### PR DESCRIPTION
**Pull request by jyegerlehner**
This is the successor to [PR 39](https://github.com/ibab/tensorflow-wavenet/pull/39). It allows one to specify use_biases: true in wavenet_params.json to enable them.

Training with and without biases didn't show a material difference in the loss vs time in the experiment I did. 

I defaulted use_biases to false in wavenet_params.json.

### CODE REVIEW

File: wavenet.py [hunk 1]
```diff
@@ -51,16 +52,33 @@ def _create_dilation_layer(self, input_batch, layer_index, dilation, in_channels
...
+          biases_gate = tf.Variable(tf.constant(0.0, shape=[dilation_channels]),
+                                  name="gate_biases")
+          conv_filter = tf.add(conv_filter, biases_filter)
+          conv_gate = tf.add(conv_gate, biases_gate)
+
         out = tf.tanh(conv_filter) * tf.sigmoid(conv_gate)
 
         weights_dense = tf.Variable(tf.truncated_normal(
             [1, dilation_channels, in_channels], stddev=0.2, name="dense"))
+
```
Comment (jyegerlehner) [inline wavenet.py]: Gratuitous white-space change.

File: wavenet.py [hunk 2]
```diff
@@ -51,16 +52,33 @@ def _create_dilation_layer(self, input_batch, layer_index, dilation, in_channels
         conv_filter = causal_conv(input_batch, weights_filter, dilation)
         conv_gate = causal_conv(input_batch, weights_gate, dilation)
 
+        if self.use_biases:
+          biases_filter = tf.Variable(tf.constant(0.0, shape=[dilation_channels]),
```
Comment (mecab) [inline wavenet.py]: As 4-spaces is used for other parts, I suggest to use it instead of 2-spaces for indentation.

File: wavenet.py [hunk 3]
```diff
@@ -51,16 +52,33 @@ def _create_dilation_layer(self, input_batch, layer_index, dilation, in_channels
...
+                                    name="filter_biases")
+          biases_gate = tf.Variable(tf.constant(0.0, shape=[dilation_channels]),
+                                  name="gate_biases")
```
Comment (mecab) [inline wavenet.py]: It could be better by align the starting to the first parameter:

```python
biases_gate = tf.Variable(tf.constant(0.0, shape=[dilation_channels]),
                          name="gate_biases")
```

**Comment by jyegerlehner** [comment]
Thanks @mecab for your comments! I think I fixed what you referred to in your comments.

**Comment by ibab** [comment]
This looks good to me!
Thanks for adding this.
